docarray/array/storage/redis/seqlike.py

```diff
@@ -0,0 +1,91 @@
+from typing import Iterable, Union
+
+from .... import Document
+from ..base.seqlike import BaseSequenceLikeMixin
+
+
+class SequenceLikeMixin(BaseSequenceLikeMixin):
+    """Implement sequence-like methods for DocumentArray with Redis as storage"""
+
+    def __eq__(self, other):
+        """Compare this object to the other, returns True if and only if other
+        as the same type as self and other has the same meta information
+
+        :param other: the other object to check for equality
+        :return: ``True`` if other is equal to self
+        """
+        # two DA are considered as the same if they have the same client meta data
+        return (
+            type(self) is type(other)
+            and self._client.client_info() == other._client.client_info()
+            and self._config == other._config
+        )
+
+    def __len__(self):
+        """Return the length of :class:`DocumentArray` that uses Redis as storage
+
+        :return: the length of this :class:`DocumentArrayRedis` object
+        """
+        try:
+            # TODO
+            # method 1
+            # keys = self._client.keys(pattern) and add same prefix to all docs in one docarray
+            # if self._offset2id_key.encode() in keys:
+            #     return len(keys) - 1
+            # else:
+            #     return len(keys)
+
+            # method 2
+            # this way, extend(), insert() funcs have to call self._save_offset2ids()
+            # if self._client.exists(self._offset2id_key.encode()):
+            #     print('offset2id exists')
+            #     return self._client.llen(self._offset2id_key.encode())
+            # else:
+            #     return 0
+
+            # method 3
+            return len(self._offset2ids)
```

Comment:
any thoughts around these different ways of implementing this function? @alaeddine-13

Reply:
@AnneYang720 not possible to do something like `self._client.keys() - 1` (or minus whatever is the number of non-document array keys) ?
If possible, I'd go with it since it reflects most up-to-date number, otherwise, len(self._offset2ids)

Reply:
I didn't find any clean commands that can return only the number of all matching keys.
There is one possible solution, using sever-side lua scipt, from [here](https://stackoverflow.com/a/26863445)

Reply:
let's keep it like this and create a follow up ticket to research more.